content/blog/2020/2020-10-01-document-embedders.md

```diff
@@ -0,0 +1,73 @@
+- **Most of the preprocessing is not required** In case of Bag of Words approach, we solve the dimensionality problem with the text preprocessing where we remove tokens (e.g. words) that seems to be less important for the analysis. It can also cause the removal of some important tokens. When using embedders, we do not need to remove tokens, so we are not losing the accuracy this way. Also most of the basic preprocessing can be omitted (such as normalization) in case of FastText embeddings.
+
+The shortcoming of the embedders is that they are difficult to understand. For example, when we use a bag-of-words, we can easily observe which tokens are important for classification with [Nomogram widget](https://orange.biolab.si/widget-catalog/visualize/nomogram/) since tokens itself are features. In the case of document embeddings, features are numbers which are not understandable to human by itself.
+
+## Document Embedding widget
+
+Orange now offers document embedders through Document Embedding widget. We decided to use [FastText pretrained embedders](https://fasttext.cc/docs/en/crawl-vectors.html), which support 157 languages. Orange's Document Embedding widget currently supports 31 most common languages.
+
+{{% workflow-screenshot src="/blog_img/2020/2020-10-01-document-embedding-widget.png" %}}
+
+In the widget, the user sets the language of documents and the aggregation method -- it is how embeddings aggregate embeddings for each word in document embedding. 
+
+## The Fake News dataset
+
+For this tutorial, we use the sample of [Fake News datast](). The dataset sample is available at [Orange's file repository](). The dataset is split in two sets training set including 2725 text items and testing set with 275 items. Each item is an article which is labelled as a real or fake.
+
+## Fake news identification 
+
+
+Here we present a fake news identification. First, we will load a training part of the dataset with the Corpus viewer.
+
+{{% workflow-screenshot src="/blog_img/2020/2020-20-01-corpus.png" %}}
+
+After the dataset is loaded, we make sure that the `text` variable is selected in the Used text features field. It means that the text in this variable is used in the text analysis. When the dataset is loaded we connect the Corpus widget to the Document embedder widget which will compute text embeddings. Our workflow should look like this now:
+
+{{% workflow-screenshot src="/blog_img/2020/2020-10-02-workflow1.png" %}}
+
+In the document embeddings widget, we check that language is set to English since texts in this datasets are English. We will use mean (average) aggregation in this experiment -- it is the most standard one which usually performs the best. After minute documents are embedded -- embedding progress is shown with the bar around the widget.
```

Comment:
I'm not sure we can claim that the mean aggregation usually gives the best performance.

Reply:
Ok, I removed it. Do you have any other reason why should we use this one? The sum is not ok because the text has a much different number of tokens. Others two would probably do as well.